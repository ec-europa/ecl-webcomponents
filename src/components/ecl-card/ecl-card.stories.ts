export default {
  title: 'Components/card',
};

const Template = (args) => 
  `<ecl-card
    theme="${args.theme}"
    image-alt="Image alternative"
    image="https://inno-ecl.s3.amazonaws.com/media/example.htmls/example.html-image.jpg"
  >
    <ecl-content-block
      theme="${args.theme}"
      meta-primary='["META1", "Meta2"]'
      meta-secondary='[{"label": "2018/10/22", "icon": "calendar"}, {"label": "Luxembourg", "icon": "location"}]'
      has-title
      has-description
      has-labels
    >
      <ecl-label
        slot="labels"
        variant="highlight"
        theme="${args.theme}"
      >
        Highlighted
      </ecl-label>
      <ecl-label
        slot="labels"
        variant="high"
        theme="${args.theme}"
      >
        Call status: Upcoming
      </ecl-label>
      <p slot="description">Transparently designing and evaluating evidence-based EU legislation, backed by citizens views.</p>
    </ecl-content-block>
  </ecl-card>`;

export const Card = Template.bind({});
Card.storyName = 'default';
