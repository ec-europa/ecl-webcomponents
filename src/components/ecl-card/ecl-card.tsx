import { Component, Prop, h } from '@stencil/core';

@Component({
  tag: 'ecl-card',
  styleUrls: {
    ec: './build/styles/ecl-card-ec.css',
    eu: './build/styles/ecl-card-eu.css',
  },
  shadow: false,
  scoped: true,
})

export class EclCard {
  @Prop() styleClass: string = '';
  @Prop() theme: string = 'ec';
  @Prop() image: string;
  @Prop() imageAlt: string;

  getClass(): string {
    return [
      `ecl-card`,
      this.styleClass
    ].join(' ');
  }

  getImgAttr() {
    const attrs = {
      class: 'ecl-card__image',
      role: 'img',
    }

    if (this.imageAlt) {
      attrs['alt'] = this.imageAlt;
    }

    return attrs;
  }

  render() {
    return (
     <article class={this.getClass()}>
      <div
        style={{ backgroundImage: "url(" + this.image + ")"}}
        {...this.getImgAttr()}
      ></div>
      <div class="ecl-card__body">
        <slot></slot>
      </div>
     </article>
    )
  }
}
