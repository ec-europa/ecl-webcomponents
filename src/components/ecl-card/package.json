{
  "name": "@ecl/web-components-card",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.6.0",
  "description": "ECL Content Item",
  "main": "ecl-card.tsx",
  "private": true,
  "scripts": {
    "build:styles": "cross-env ecl-builder styles",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.8.4",
    "@ecl/vanilla-component-card": "3.8.4",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
