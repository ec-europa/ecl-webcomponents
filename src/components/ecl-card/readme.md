# ecl-card



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description | Type     | Default     |
| ------------ | ------------- | ----------- | -------- | ----------- |
| `image`      | `image`       |             | `string` | `undefined` |
| `imageAlt`   | `image-alt`   |             | `string` | `undefined` |
| `styleClass` | `style-class` |             | `string` | `''`        |
| `theme`      | `theme`       |             | `string` | `'ec'`      |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
