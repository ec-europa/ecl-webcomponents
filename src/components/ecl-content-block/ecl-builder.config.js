const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  scripts: [
    {
      entry: path.resolve(nodeModules, '@ecl/vanilla-component-content-block/content-block.js'),
      dest: path.resolve(outputFolder, 'scripts/ecl-content-block-vanilla.js'),
      options: {
        banner,
        moduleName: 'ECL',
        includePaths,
        sourceMap: false,
      },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-content-block-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-content-block-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-content-block-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-content-block-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
