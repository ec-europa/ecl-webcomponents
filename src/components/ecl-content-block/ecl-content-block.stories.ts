const getArgs = () => {
  return {
    title: 'Title',
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus gravida ipsum ut lorem cursus,
                  quis tincidunt sem viverra. Nunc vestibulum, mauris quis porta venenatis, justo odio commodo tellus`,
    metaPrimary:  '["PRIMARY META", "DD Month Year"]',
    metaSecondary: '[{"label": "2018/10/22", "icon": "calendar"}, {"label": "Luxembourg", "icon": "location"}]',
  };
};

const getArgTypes = () => {
  return {
    title: {
      type: { name: 'string'},
      description: 'Title of the content block',
    },
    description: {
      type: { name: 'string'},
      description: 'Description',
    },
    metaPrimary: {
      name: "Primary meta",
      type: { name: 'string'},
      description: 'Primary meta element',
    },
    metaSecondary: {
      name: 'Secondary meta',
      type: { name: 'string'},
      description: 'Secondary meta element',
    },
  };
};

export default {
  title: 'Components/content-block',
};

const Template = (args) => 
  `<ecl-content-block
    theme="${args.theme}"
    meta-primary='${args.metaPrimary}'
    meta-secondary='${args.metaSecondary}'
    has-title
    has-description
    has-lists
    has-labels
    has-links
    has-secondary-links
  >
    <ecl-label
      slot="labels"
      variant="highlight"
      theme="${args.theme}"
    >
      Highlight
    </ecl-label>
    <ecl-label
      slot="labels"
      variant="high"
      theme="${args.theme}"
    >
      High importance
    </ecl-label>
    <ecl-text tag="p" slot="description">${args.description}</ecl-tag>
    <ecl-link
      theme="${args.theme}"
      slot="title"
      path="/example.html"
      variant="standalone"
    >
      ${args.title}
    </ecl-link>
    <ecl-link
      slot="links"
      theme="${args.theme}"
      variant="standalone"
      path="/example.html"
    >
      Primary Link 1
    </ecl-link>
    <ecl-link
      slot="links"
      theme="${args.theme}"
      variant="standalone"
      path="/example.html"
    >
      Primary Link 2
    </ecl-link>
    <ecl-link
      slot="links"
      theme="${args.theme}"
      variant="standalone"
      path="/example.html"
    >
      Primary Link 3
    </ecl-link>
    <ecl-link
      slot="links"
      theme="${args.theme}"
      variant="standalone"
      path="/example.html"
    >
      Primary Link 4
    </ecl-link>
    <ecl-link
      slot="links-secondary"
      theme="${args.theme}"
      variant="standalone"
      path="/example.html"
    >
      Secondary Link 1
    </ecl-link>
    <ecl-link
      slot="links-secondary"
      theme="${args.theme}"
      variant="standalone"
      path="/example.html"
    >
      Secondary Link 2
    </ecl-link>
    <ecl-description-list
      theme="${args.theme}"
      slot="lists"
    >
      <ecl-description-list-term
        theme="${args.theme}"
      >
        Standard text
      </ecl-description-list-term>
      <ecl-description-list-definition
        type="text"
        theme="${args.theme}"
      >
        Lorem ipsum dolor sit amet, <a href="/component-library/example" class="ecl-link">consectetur adipiscing elit</a>. Suspendisse ut sapien condimentum, aliquet turpis sit amet, finibus purus. Donec porttitor iaculis felis ut dapibus. Sed blandit, massa ac suscipit facilisis
      </ecl-description-list-definition>
      <ecl-description-list-term
        theme="${args.theme}"
      >
        Standalone links
      </ecl-description-list-term>
      <ecl-description-list-definition
        type="link"
        theme="${args.theme}"
        items='[{"label": "Lorem ipsum dolor sit amet", "path": "/example.html", "icon": "copy"},{"label": "Lorem ipsum dolor sit amet", "path": "/example.html", "icon": "download"}]'
      >
      </ecl-description-list-definition>
      <ecl-description-list-term
        theme="${args.theme}"
      >
        Links inline
      </ecl-description-list-term>
      <ecl-description-list-definition
        type="inline"
        theme="${args.theme}"
        items='[{"label": "Lorem ipsum dolor sit amet", "path": "/example.html"},{"label": "Lorem ipsum dolor sit amet", "path": "/example.html"}]'
      >
      </ecl-description-list-definition>
      <ecl-description-list-term
        theme="${args.theme}"
      >
        Taxonomy list
      </ecl-description-list-term>
      <ecl-description-list-definition
        type="taxonomy"
        theme="${args.theme}"
        items='["Taxonomy item 1",{"label": "Taxonomy item 2", "path": "/example.html"},"Taxonomy item 3"]'
      ></ecl-description-list-definition>
    </ecl-description-list>
  </ecl-content-block>`;

export const ContentBlock = Template.bind({});
ContentBlock.storyName = 'default';
ContentBlock.args = getArgs();
ContentBlock.argTypes = getArgTypes();
