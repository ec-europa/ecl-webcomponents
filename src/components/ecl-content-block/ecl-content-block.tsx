import { Component, Prop, h, Element } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";
declare const ECL: any;

@Component({
  tag: 'ecl-content-block',
  styleUrls: {
    ec: './build/styles/ecl-content-block-ec.css',
    eu: './build/styles/ecl-content-block-eu.css',
  },
  assetsDirs: ['build'],
  shadow: false,
  scoped: true,
})

export class EclContentBlock {
  @Element() el: HTMLElement;
  @Prop() styleClass: string = '';
  @Prop() theme: string = 'ec';
  @Prop() hasDescription: boolean;
  @Prop() eclScript: boolean = false;
  @Prop() hasTitle: boolean;
  @Prop() hasLabels: boolean;
  @Prop() hasLinks: boolean;
  @Prop() hasSecondaryLinks: boolean;
  @Prop() hasLists: boolean;
  @Prop() metaPrimary: string;
  @Prop() metaSecondary: string;

  getClass(): string {
    return [
      `ecl-content-block`,
      this.styleClass
    ].join(' ');
  }

  componentDidRender() {
    if (this.hasLabels) {
      const labelsContainer = this.el.querySelector('.ecl-content-block__label-container');
      const labels = this.el.querySelectorAll('.ecl-label');
      if (labels) {
        labelsContainer.innerHTML = '';
        labels.forEach((label) => {
          label.classList.add(`sc-ecl-content-block-${this.theme}`);
          const listEl = document.createElement('li');
          listEl.classList.add('ecl-content-block__label-item', `sc-ecl-content-block-${this.theme}`);
          listEl.appendChild(label);
          labelsContainer.appendChild(listEl);
        });
      }
    }
    if (this.hasLinks) {
      const linksContainers = this.el.querySelectorAll('.ecl-content-block__link-list');

      linksContainers.forEach((linkList) => {
        const thisLinks = linkList.querySelectorAll('.ecl-link');
        if (thisLinks) {
          linkList.innerHTML = '';
          thisLinks.forEach((link) => {
            link.classList.add(`sc-ecl-content-block-${this.theme}`);
            const listEl = document.createElement('li');
            listEl.classList.add('ecl-content-block__link-item', `sc-ecl-content-block-${this.theme}`);
            listEl.appendChild(link);
            linkList.appendChild(listEl);
          });
        }
      });
    }
    if (this.hasLists) {
      const list = this.el.querySelector('.ecl-content-block__list-container .ecl-description-list');
      if (list) {
        list.classList.add('ecl-content-block__list', `sc-ecl-content-block-${this.theme}`);
      }
    }

    if (this.eclScript) {
      // Load the ECL vanilla js if not already present.
      const src = getAssetPath('./build/scripts/ecl-content-block-vanilla.js');
      if (document.querySelector(`script[src="${src}"]`)) {
        document.querySelector(`script[src="${src}"]`).remove();
      }
      const script = document.createElement('script');
      script.src = src;
      script.onload = () => {
        const contentBlock = new ECL.ContentBlock(this.el);
        contentBlock.init();
      };

      document.body.appendChild(script);
    }
  }

  render() {
    const metaPrimaryArray = this.metaPrimary ? JSON.parse(this.metaPrimary) : '';
    const metaSecondaryArray = this.metaSecondary ? JSON.parse(this.metaSecondary) : '';
    return (
     <div class={this.getClass()}>
      { this.hasLabels ?
        <ul class="ecl-content-block__label-container">
          <slot name="labels"></slot>
        </ul> : ''
      }
      { metaPrimaryArray ?
        <ul class="ecl-content-block__primary-meta-container">
        { metaPrimaryArray.map((meta) => (
          <li class="ecl-content-block__primary-meta-item">
            {meta}
          </li>
          ))
        }  
        </ul> : '' 
      }
      { this.hasTitle ?
        <h1 class="ecl-content-block__title">
          <slot name="title"></slot>
        </h1> : ''
      }
      { this.hasDescription ?
        <div class="ecl-content-block__description">
          <slot name="description"></slot>
        </div> : ''
      }
      { metaSecondaryArray ?
        <ul class="ecl-content-block__secondary-meta-container">
        { metaSecondaryArray.map((meta) => (
          <li class="ecl-content-block__secondary-meta-item">
          { meta.icon ?
            <ecl-icon
              theme={this.theme}
              icon={meta.icon}
              size="s"
              style-class={`ecl-content-block__secondary-meta-icon sc-ecl-content-block-${this.theme}`}
            ></ecl-icon> : ''
          }
          { meta.label ? 
            <span class="ecl-content-block__secondary-meta-label">
              {meta.label}
            </span> : ''
          }
          </li>
          ))
        }  
        </ul> : '' 
      }
      { this.hasLinks ?
        <div class="ecl-content-block__link-container">
          <ul class="ecl-content-block__link-list">
            <slot name="links"></slot>
          </ul>
          { this.hasSecondaryLinks ?
          <ul class="ecl-content-block__link-list">
            <slot name="links-secondary"></slot>
          </ul> : ''
          }
        </div> : ''
      }
      { this.hasLists ?
        <div class="ecl-content-block__list-container">
          <slot name="lists"></slot>
        </div> : ''
      }
     </div> 
    )
  }
}
