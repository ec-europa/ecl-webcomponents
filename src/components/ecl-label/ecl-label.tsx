import { Component, Prop, h } from '@stencil/core';

@Component({
  tag: 'ecl-label',
  styleUrls: {
    ec: './build/styles/ecl-label-ec.css',
    eu: './build/styles/ecl-label-eu.css',
  },
  shadow: false,
  scoped: true,
})

export class EclLabel {
  @Prop() styleClass: string = '';
  @Prop() variant: string = 'low';
  @Prop({ mutable: true }) theme: string;

  getClass(): string {
    return [
      `ecl-label`,
      `ecl-label--${this.variant}`,
      this.styleClass
    ].join(' ');
  }

  componentWillLoad() {
    this.theme = document.documentElement.getAttribute('data-ecl-theme') ?? (this.theme || 'ec');
  }

  render() {
    return (
      <span class={this.getClass()}>
        <slot></slot>
      </span>
    )
  }
}
