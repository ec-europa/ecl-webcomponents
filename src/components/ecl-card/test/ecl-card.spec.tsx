import { newSpecPage } from '@stencil/core/testing';
import { EclCard } from '../ecl-card';
import { EclContentBlock } from '../../ecl-content-block/ecl-content-block';
import { EclLabel } from '../../ecl-label/ecl-label';

describe('ecl-card', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclCard, EclLabel, EclContentBlock],
      html: `<ecl-card
      theme="ec"
      image-alt="Image alternative"
      image="https://inno-ecl.s3.amazonaws.com/media/examples/example.html-image.jpg"
    >
    <ecl-content-block
      theme="ec"
      meta-primary='["META1", "Meta2"]'
      meta-secondary='[{"label": "2018/10/22", "icon": "calendar"}, {"label": "Luxembourg", "icon": "location"}]'
      has-title
      has-description
      has-labels
    >
      <ecl-label
        slot="labels"
        variant="highlight"
        theme="ec"
      >
        Highlighted
      </ecl-label>
      <ecl-label
        slot="labels"
        variant="high"
        theme="ec"
      >
        Call status: Upcoming
      </ecl-label>
      <p slot="description">Transparently designing and evaluating evidence-based EU legislation, backed by citizens views.</p>
    </ecl-content-block>
  </ecl-card>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
