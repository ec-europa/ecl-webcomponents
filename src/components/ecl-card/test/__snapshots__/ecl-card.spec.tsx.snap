// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-card renders 1`] = `
<ecl-card image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg" image-alt="Image alternative" theme="ec">
  <!---->
  <article class="ecl-card">
    <ecl-picture image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg" imagealt="Image alternative" imgclass="ecl-card__image sc-ecl-card-ec" styleclass="ecl-card__picture sc-ecl-picture-ec"></ecl-picture>
    <div class="ecl-card__body">
      <ecl-content-block has-description="" has-labels="" has-title="" meta-primary="[&quot;META1&quot;, &quot;Meta2&quot;]" meta-secondary="[{&quot;label&quot;: &quot;2018/10/22&quot;, &quot;icon&quot;: &quot;calendar&quot;}, {&quot;label&quot;: &quot;Luxembourg&quot;, &quot;icon&quot;: &quot;location&quot;}]" theme="ec">
        <!---->
        <div class="ecl-content-block">
          <ul class="ecl-content-block__label-container">
            <li class="ecl-content-block__label-item sc-ecl-content-block-ec">
              <span class="ecl-label ecl-label--highlight sc-ecl-content-block-ec">
                Highlighted
              </span>
            </li>
            <li class="ecl-content-block__label-item sc-ecl-content-block-ec">
              <span class="ecl-label ecl-label--high sc-ecl-content-block-ec">
                Call status: Upcoming
              </span>
            </li>
          </ul>
          <ul class="ecl-content-block__primary-meta-container">
            <li class="ecl-content-block__primary-meta-item">
              META1
            </li>
            <li class="ecl-content-block__primary-meta-item">
              Meta2
            </li>
          </ul>
          <h1 class="ecl-content-block__title"></h1>
          <div class="ecl-content-block__description">
            <p slot="description">
              Transparently designing and evaluating evidence-based EU legislation, backed by citizens views.
            </p>
          </div>
          <ul class="ecl-content-block__secondary-meta-container">
            <li class="ecl-content-block__secondary-meta-item">
              <ecl-icon icon="calendar" size="s" style-class="ecl-content-block__secondary-meta-icon sc-ecl-content-block-ec" theme="ec"></ecl-icon>
              <span class="ecl-content-block__secondary-meta-label">
                2018/10/22
              </span>
            </li>
            <li class="ecl-content-block__secondary-meta-item">
              <ecl-icon icon="location" size="s" style-class="ecl-content-block__secondary-meta-icon sc-ecl-content-block-ec" theme="ec"></ecl-icon>
              <span class="ecl-content-block__secondary-meta-label">
                Luxembourg
              </span>
            </li>
          </ul>
        </div>
      </ecl-content-block>
    </div>
  </article>
</ecl-card>
`;
